.github/workflows/generate-preview-link.yml

```diff
@@ -0,0 +1,134 @@
+name: Generate preview link
+
+permissions:
+    actions: write
+    checks: write
+    contents: write
+    deployments: write
+    pull-requests: write
+    statuses: write
+
+on:
+    workflow_run:
+        workflows: ['Pre-generate preview link']
+        types:
+            - completed
+
+env:
+    NODE_OPTIONS: '--max-old-space-size=8192'
+
+concurrency:
+    group: cloudflare-pages-build-${{ github.event.workflow_run.head_branch }}
+    cancel-in-progress: true
+
+jobs:
+    build_to_cloudflare_pages:
+        runs-on: Ubuntu-latest
+        if: >
+            github.event.workflow_run.event == 'pull_request' &&
+            github.event.workflow_run.conclusion == 'success'
+        steps:
+            - name: Download artifact
+              id: download-artifact
+              uses: dawidd6/action-download-artifact@v2
+              with:
+                  workflow_conclusion: success
+                  run_id: ${{ github.event.workflow_run.id }}
+                  name: 'pr-${{ github.event.workflow_run.id }}'
+
+            - name: Retrieve and verify user organization
+              id: pr_information
+              run: |
+                  echo "Verifying user's organization..."
+                  user=$(cat ./USERNAME)
+                  response=$(curl -s -L \
+                      -w "%{http_code}" \
+                     -o /dev/null -H "Accept: application/vnd.github+json" \
+                     -H "Authorization: Bearer ${{ secrets.LIST_ORGS_TOKEN }}" \
```

Comment:
We need a separate Github classic token for this with SSO verification for binary-com organization
cc: @balakrishna-deriv

Reply:
Will github.token work? I think it might work because we are only doing a read operation.

Reply:
Noo it needs a new token that is SSOed with binary-com organization, the GITHUB_TOKEN is provided by default

Reply:
<img width="828" alt="Screenshot 2023-09-25 at 12 14 51 PM" src="https://github.com/binary-com/deriv-com/assets/103016120/3d035f6a-8b2f-4415-80c3-ada7e79d0cb6">